Binary Search Trees/Binary Search Tree Implementation Complete.py

```diff
@@ -0,0 +1,202 @@
+class Node:
+    def __init__(self,value):
+        self.value = value
+        self.left = None
+        self.right = None
+    
+    def insert(self, value):
+        if self.value == value:
+            return
+        if self.value > value:
+            if self.left:
+                return self.left.insert(value)
+            else:
+                self.left = Node(value)
+        if self.value < value:
+            if self.right:
+                return self.right.insert(value)
+            else:
+                self.right = Node(value)
+    
+    def find(self, value):
+        found = False
+        if self.value == value: #check to see if the current node's value matches the search value. NOTE - initiation self == root.
+            found = True
+        elif self.value == None:
+            found = False
+        elif self.value < value: #if the value is greater, check the right side of the tree if right side is there.
+            if self.right:
+                return self.right.find(value)
```

Comment:
From this line, it seems like find method returns some value and that value is being returned from here, but find actually doesn't return anything, right? I think the more appropriate options are:
1. `self.right.find(value)`, then `return` in the next line.
2. Make find return a value (True, False, or the Node itself) and keep this line as is.
I think number 2 is overall the better option, so the callers of find can use the node in some way. In that case, we wouldn't need the `found` variable anymore.

Reply:
Oh, I tried to keep track while traversing through the BST. As long as I couldn't find it, found was set to false and traversal went on. Once the value was found, found was set to True. Depending on found or not, a line would be printed out saying, "the value, value, is in the tree".